services/tracing/tracing/application.go

```diff
@@ -0,0 +1,44 @@
+package tracing
+
+import "sensorbucket.nl/sensorbucket/pkg/pipeline"
+
+type StepStore interface {
+	Insert(Step) error
+}
+
+func New(stepStore StepStore) *Service {
+	return &Service{
+		stepStore: stepStore,
+	}
+}
+
+type Service struct {
+	stepStore StepStore
+}
+
+func (s *Service) HandlePipelineMessage(pipelineMessage pipeline.Message) error {
+	return s.addStep(Step{
+		TracingID: pipelineMessage.ID,
+		StepIndex: pipelineMessage.StepIndex,
+
+		// We have to add 1 to the stepindex to get the actual steps remaining
+		StepsRemaining: int64(len(pipelineMessage.PipelineSteps) - (int(pipelineMessage.StepIndex + 1))),
+
+		// The timestamp is set by the mq when it is send to the queue. The next step's starttime can be used to deduce the processing time between the 2 steps
```

Comment:
Small comment addition: This duration consists of: Time in Queue and Processing Time